agent/03_agent_deployment.sh

```diff
@@ -134,10 +208,40 @@ function attach_fleeting_iso() {
 
 write_pull_secret
 
-node0_ip=$(nth_ip $EXTERNAL_SUBNET_V4 20)
+# needed for assisted-service to run nmstatectl
+# This is temporary and will go away when https://github.com/nmstate/nmstate is used
+sudo yum install -y nmstate
+
+if [[ "$FLEETING_STATIC_IP_NODE0_ONLY" = "true" ]]; then
+    static_ips=1
+else
+    static_ips=$NUM_MASTERS+$NUM_WORKERS
+fi
+
+if [[ "$IP_STACK" = "v4" ]]; then
+    external_subnet=$EXTERNAL_SUBNET_V4
+else
+    external_subnet=$EXTERNAL_SUBNET_V6
+fi
+
+# Use an IP address outside of the range that is set for DHCP, just for sanity checks.
+# It shouldn't matter as the dhcp addresses will not be picked up when using static IPs
+base_ip=40
+for (( n=0; n<${static_ips}; n++ ))
+do
+    ip=${base_ip}+${n}
+    ip_list+=($(nth_ip ${external_subnet} ${ip}))
+done
+
 set_api_and_ingress_vip
 
-generate_fleeting_iso ${node0_ip}
+mac_list=()
+set_mac_addresses master $NUM_MASTERS
+set_mac_addresses worker $NUM_WORKERS
```

Comment:
I think there could be some space to improve the management of the static ips and mac lists. The two lists are synchronized, as their nth index refers to the same host, but there's no clue about that in the name. Also, their creation is performed in separate places (and with different approaches), and I think that this could make easier to introduce some errors in future.
I'd suggest to:
1) Use a more consistent namings for them, ie something like global vars `FLEETING_NODES_IPS` and `FLEETING_NODES_MACS` (btw simplify also the parameters passing)
2) Have just one function for generating both of them in the first level of the script
3) Some strong assumption about ordering are made elsewhere (ie first masters and then workers). That's reasonable when dealing with a single list, but please let's document it to avoid missing it in future

Reply:
Good points @andfasano. I changed to use globals and reworked it to use one function to set everything up to ensure consistency.